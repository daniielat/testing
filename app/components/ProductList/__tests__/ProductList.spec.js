const React = require('react');
const ProductList = require('..');
const { render, screen, fireEvent } = require('@testing-library/react');
const mockProducts = require('../../../../mocks/test/get/https/api.mercadolibre.com/sites/MLA/search?limit=10&offset=0&q=celular.json');

describe('El componente Pagination', () => {
    let component;
    const i18n = { gettext: text => text };
    const setProductList = jest.fn();
    
    beforeEach(() => {
        component = render(<ProductList i18n={i18n} products={mockProducts.data.data} setProductList={setProductList}/>);
    });

    it('Renderiza', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();
    });

    // it('Aumenta el valor de offser al clickear el botÃ³n `Siguiente`', () => {
    //     const button = screen.getByRole('button');
    //     fireEvent.click(button);

    //     expect(offset).toEqual(10);
    // });
});