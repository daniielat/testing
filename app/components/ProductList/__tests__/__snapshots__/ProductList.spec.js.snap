// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`El componente Pagination Renderiza 1`] = `
<DocumentFragment>
  <section>
    <h2>
      Listado de productos
    </h2>
    <section>
      <button
        class="andes-button andes-button--large andes-button--quiet"
        type="button"
      >
        <span
          class="andes-button__content"
        >
          Siguiente
        </span>
      </button>
    </section>
    <h2>
      No se encontraron productos
    </h2>
  </section>
</DocumentFragment>
`;
